Marlin/pins_MEGACONTROLLER.h

```diff
@@ -0,0 +1,132 @@
+/**
+ * Mega controller pin assignments
+ */
+
+#ifndef __AVR_ATmega2560__
+  #error Oops!  Make sure you have 'Arduino Mega' selected from the 'Tools -> Boards' menu.
+#endif
+
+#if EXTRUDERS > 3
+  #error RUMBA supports up to 3 extruders. Comment this line to keep going.
+#endif
+ #define X_STEP_PIN 62//A8
+ #define X_DIR_PIN 63//A9
+ #define X_ENABLE_PIN 61//A7
+#ifdef DELTA
+ #define X_MIN_PIN -1
+#else
+ #define X_MIN_PIN 43
+#endif //DELTA
+ #define X_MAX_PIN 42 //2 //Max endstops default to disabled "-1", set to commented value to enable.
+
+ #define Y_STEP_PIN 65 // A11
+ #define Y_DIR_PIN 66 // A12
+ #define Y_ENABLE_PIN 64//A10
+#ifdef DELTA
+ #define Y_MIN_PIN -1
+#else
+ #define Y_MIN_PIN 38
+#endif //DELTA
```

Comment:
You don't need to test `DELTA` in your pins file. A `DELTA` will have no min endstops, and the lack of min endstops will cause these pins to be set to -1 later in `pins.h`. So, just set this to 38 and `X_MIN_PIN` to 43.